### PR TITLE
Multi tenancy broker adjustments ido

### PR DESCRIPTION
**Pull request by idonaaman123**

### CODE REVIEW

File: db/db.go [hunk 1]
```diff
@@ -37,6 +37,8 @@ var configuration = conf.GetConfig()
 
 var MetadataDbClient MetadataStorage
 
+var GlobalTenantName = "1"
```
Comment (shay23b) [inline db/db.go:40]: remove

File: db/db.go [hunk 2]
```diff
@@ -928,7 +962,7 @@ func GetActiveStations() ([]models.Station, error) {
 		return []models.Station{}, err
 	}
 	defer conn.Release()
-	query := `SELECT * FROM stations AS s WHERE s.is_deleted = false OR s.is_deleted IS NULL`
+	query := `SELECT * FROM stations AS s WHERE (s.is_deleted = false OR s.is_deleted IS NULL)`
```
Comment (shay23b) [inline db/db.go:965]: why did you add the parenthesis?

File: db/db.go [hunk 3]
```diff
@@ -1107,12 +1143,168 @@ func InsertNewStation(
...
+// TODO: check if need to add condition instead duplicate same functions
+func GetAllStationsDetailsPerTenant(tenantName string) ([]models.ExtendedStation, error) {
+	ctx, cancelfunc := context.WithTimeout(context.Background(), DbOperationTimeout*time.Second)
+	defer cancelfunc()
+	conn, err := MetadataDbClient.Client.Acquire(ctx)
+	if err != nil {
+		return []models.ExtendedStation{}, err
+	}
+	defer conn.Release()
+	query := `
```
Comment (shay23b) [inline db/db.go:1162]: tenant name is not in the query - add it to where section

File: db/db.go [hunk 4]
```diff
@@ -1473,42 +1667,42 @@ func GetStationNamesUsingSchema(schemaName string) ([]string, error) {
...
+func RemoveSchemaFromAllUsingStations(schemaName string, tenantName string) error {
 	ctx, cancelfunc := context.WithTimeout(context.Background(), DbOperationTimeout*time.Second)
 	defer cancelfunc()
 	conn, err := MetadataDbClient.Client.Acquire(ctx)
 	if err != nil {
 		return err
 	}
 	defer conn.Release()
-	query := `UPDATE stations SET schema_name = '' WHERE schema_name = $1`
+	query := `UPDATE stations SET schema_name = '' WHERE schema_name = $1 and tenant_name=$2`
```
Comment (shay23b) [inline db/db.go:1700]: and -> AND

File: db/db.go [hunk 5]
```diff
@@ -3161,7 +3354,7 @@ func CreateUser(username string, userType string, hashedPassword string, fullNam
 		full_name, 
 		subscription,
 		skip_get_started,
-		tenant_name) 
+		tenant_id)
```
Comment (shay23b) [inline db/db.go:3357]: why did you change tenant name to tenant id?

File: db/db.go [hunk 6]
```diff
@@ -3224,7 +3418,7 @@ func ChangeUserPassword(username string, hashedPassword string, tenantName strin
 		return err
 	}
 	defer conn.Release()
-	query := `UPDATE users SET password = $2 WHERE username = $1 AND tenant_name = $3`
+	query := `UPDATE users SET password = $2 WHERE username = $1`
```
Comment (shay23b) [inline db/db.go:3421]: why did you remove tenant name?

File: db/db.go [hunk 7]
```diff
@@ -3244,7 +3438,7 @@ func GetRootUser(tenantName string) (bool, models.User, error) {
 		return false, models.User{}, err
 	}
 	defer conn.Release()
-	query := `SELECT * FROM users WHERE type = 'root' AND tenant_name = $1 LIMIT 1`
+	query := `SELECT * FROM users WHERE type = 'root' LIMIT 1`
```
Comment (shay23b) [inline db/db.go:3441]: why did you remove tenant name?

File: db/db.go (hunk 3 above)
Comment (idonaaman123) [reply db/db.go:1162]: its not should be, the s.* retrieve it

File: server/memphis_handlers_user_mgmt.go [hunk 8]
```diff
@@ -620,12 +631,13 @@ func (umh UserMgmtHandler) AddUser(c *gin.Context) {
...
 		return
 	}
+
 	user, err := getUserDetailsFromMiddleware(c)
 	if err != nil {
-		serv.Errorf("CreateUser: User " + body.Username + ": " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-		return
+		serv.Errorf("AddUser: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:638]: you missed return after this line

File: server/memphis_handlers_user_mgmt.go [hunk 9]
```diff
@@ -914,6 +931,12 @@ func (umh UserMgmtHandler) EditCompanyLogo(c *gin.Context) {
 		return
 	}
 
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("EditCompanyLogo: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:937]: you missed also here the return after this line. please pay attention

File: server/memphis_handlers_user_mgmt.go [hunk 10]
```diff
@@ -941,7 +964,12 @@ func (umh UserMgmtHandler) EditCompanyLogo(c *gin.Context) {
 }
 
 func (umh UserMgmtHandler) RemoveCompanyLogo(c *gin.Context) {
-	err := db.DeleteImage("company_logo")
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("RemoveCompanyLogo: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:970]: you missed also here the return after this line. please pay attention

File: server/memphis_handlers_user_mgmt.go [hunk 11]
```diff
@@ -952,7 +980,12 @@ func (umh UserMgmtHandler) RemoveCompanyLogo(c *gin.Context) {
 }
 
 func (umh UserMgmtHandler) GetCompanyLogo(c *gin.Context) {
-	exist, image, err := db.GetImage("company_logo")
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("GetCompanyLogo: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:986]: you missed also here the return after this line. please pay attention

File: server/memphis_handlers_user_mgmt.go [hunk 12]
```diff
@@ -1075,16 +1108,21 @@ func (umh UserMgmtHandler) GetFilterDetails(c *gin.Context) {
...
-		users, err := umh.GetActiveUsers()
+		users, err := umh.GetActiveUsers(strings.ToLower(user.TenantName))
 		if err != nil {
 			serv.Errorf("GetFilterDetails: GetActiveUsers: " + err.Error())
 			c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
 			return
 		}
 
-		tags, err := umh.GetActiveTags()
+		tags, err := umh.GetActiveTags(user.TenantName)
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:1125]: strings.ToLower(user.TenantName)

File: server/memphis_handlers_user_mgmt.go [hunk 13]
```diff
@@ -742,7 +744,13 @@ func (umh UserMgmtHandler) AddUser(c *gin.Context) {
 }
 
 func (umh UserMgmtHandler) GetAllUsers(c *gin.Context) {
-	users, err := db.GetAllUsers()
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("GetAllUsers: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
+	}
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:751]: add return

File: server/memphis_handlers_user_mgmt.go [hunk 14]
```diff
@@ -953,7 +962,7 @@ func (umh UserMgmtHandler) EditCompanyLogo(c *gin.Context) {
 
 	_ = os.Remove(fileName)
 
-	err = db.InsertImage("company_logo", base64Encoding, strings.ToLower(user.TenantName))
+	err = db.InsertImage("company_logo", base64Encoding, user.TenantName)
 	if err != nil {
 		serv.Errorf("EditCompanyLogo: " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_user_mgmt.go:968]: add return

File: server/memphis_handlers_tags.go [hunk 15]
```diff
@@ -146,7 +147,7 @@ func (th TagsHandler) CreateNewTag(c *gin.Context) {
 	stationArr := []int{}
 	schemaArr := []int{}
 	userArr := []int{}
-	newTag, err := db.InsertNewTag(name, color, stationArr, schemaArr, userArr, strings.ToLower(user.TenantName))
+	newTag, err := db.InsertNewTag(name, color, stationArr, schemaArr, userArr, tenantName)
 	if err != nil {
 		serv.Errorf("CreateNewTag: Tag " + body.Name + ": " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
```
Comment (shohamroditimemphis) [inline server/memphis_handlers_tags.go:153]: add return

**Review by shohamroditimemphis — APPROVED** [review]
